app/src/main/java/ch/epfl/culturequest/notifications/PushNotification.java

```diff
@@ -0,0 +1,168 @@
+     *
+     * @param context the context of the notification
+     */
+    public static void createNotificationChannels(Context context) {
+        // Create the NotificationChannels, but only on API 26+ because
+        // the NotificationChannel class is new and not in the support library
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+
+            // retrieve a list of all notifications channels
+            List<NotificationChannel> channels = new ArrayList<>();
+            channels.add(FollowNotification.getNotificationChannel());
+            channels.add(ScanNotification.getNotificationChannel());
+            channels.add(LikeNotification.getNotificationChannel());
+            channels.add(CompetitionNotification.getNotificationChannel());
+            channels.add(SightseeingNotification.getNotificationChannel());
+
+            // Register the channel with the system; you can't change the importance
+            // or other notification behaviors after this
+            NotificationManager notificationManager = getSystemService(context, NotificationManager.class);
+            Objects.requireNonNull(notificationManager).createNotificationChannels(channels);
+        }
+    }
+
+    /**
+     * Returns the notification ready to be sent
+     *
+     * @param context the context of the notification
+     * @return the notification
+     */
+    public Notification buildNotification(Context context) {
+        PendingIntent pendingIntent = selectPendingIntent(context, channelId);
+        return new NotificationCompat.Builder(context, channelId)
+                .setSmallIcon(R.drawable.logo_compact)
+                .setContentTitle(title)
+                .setContentText(text)
+                .setPriority(NotificationCompat.PRIORITY_DEFAULT)
+                .setContentIntent(pendingIntent)
+                .build();
+    }
+
+    /**
+     * Returns the pending intent associated with the notification
+     *
+     * @param context   the context of the notification
+     * @param channelId the channel id of the notification
+     * @return the pending intent
+     */
+    public PendingIntent selectPendingIntent(Context context, String channelId) {
+        switch (channelId) {
+            case FollowNotification.CHANNEL_ID:
+                return FollowNotification.getPendingIntent(context);
+            case ScanNotification.CHANNEL_ID:
+                return ScanNotification.getPendingIntent(context);
+            case LikeNotification.CHANNEL_ID:
+                return LikeNotification.getPendingIntent(context);
+            case CompetitionNotification.CHANNEL_ID:
+                return CompetitionNotification.getPendingIntent(context);
+            case SightseeingNotification.CHANNEL_ID:
+                return SightseeingNotification.getPendingIntent(context);
+            default:
+                return null;
```

Comment:
Instead of null, shouldn't it return a basic intent? One that just opens the app?